src/shared/local-addresses.c

```diff
@@ -315,6 +319,9 @@ int local_gateways(
                                 continue;
                         }
 
+                        if (af != AF_UNSPEC)
+                                continue;
```

Comment:
this doesn't look right to me? it would mean if I look for AF_INET gateways only we'd not pick the gateway up here? what am i missing?

Reply:
This may look confusing, but should be correct.
`RTA_VIA` attribute is provided for an IPv4 route with IPv6 gateway. When we request "IPv4 gateways", that is `af == AF_INET`, then I think we should return only IPv4 addresses. Also, when we request IPv6 gateways, we should return IPv6 addresses. So, `RTA_VIA` should be parsed only when `af == AF_UNSPEC`.
I will add comments around the condition.